### PR TITLE
Horizontal Pod Autoscaler is deleted along with namespace

### PR DESCRIPTION
**Pull request by socaa**
Part of #11570 

@piosz 
@mwielgus

### CODE REVIEW

**Comment by piosz** [comment]
Assigned @mwielgus who is a big fan a this attitude.

File: test/e2e/autoscaling_utils.go [hunk 1]
```diff
@@ -103,6 +103,7 @@ func (rc *ResourceConsumer) sendConsumeCPUrequests(requests, millicores, duratio
 
 // sendOneConsumeCPUrequest sends POST request for cpu consumption
 func (rc *ResourceConsumer) sendOneConsumeCPUrequest(millicores int, durationSec int) {
+	defer GinkgoRecover()
```
Comment (piosz) [inline test/e2e/autoscaling_utils.go]: why do you need to call it here?

Comment (socaa) [inline test/e2e/autoscaling_utils.go]: when err (from sending post request) is not nil, then we need defer GinkgoRecover().
sendOneConsumeCPUrequest is always in a new gorutine.

Comment (piosz) [inline test/e2e/autoscaling_utils.go]: Ok. Right

File: test/e2e/autoscaling_utils.go [hunk 2]
```diff
@@ -30,7 +30,7 @@ import (
 const (
 	consumptionTimeInSeconds = 30
 	sleepTime                = 30 * time.Second
-	requestSizeInMilicores   = 100
+	requestSizeInMillicores  = 100
```
Comment (mwielgus) [inline test/e2e/autoscaling_utils.go]: Isn't it done in a different PR?

File: test/e2e/autoscaling_utils.go [hunk 3]
```diff
@@ -142,7 +143,6 @@ func (rc *ResourceConsumer) CleanUp() {
 	rc.stop <- 0
 	expectNoError(DeleteRC(rc.framework.Client, rc.framework.Namespace.Name, rc.name))
 	expectNoError(rc.framework.Client.Services(rc.framework.Namespace.Name).Delete(rc.name))
```
Comment (mwielgus) [inline test/e2e/autoscaling_utils.go]: Where is the namespace deleted?

Comment (socaa) [inline test/e2e/autoscaling_utils.go]: namespace for suite is automatically deleted in horizontal_pod_autoscaling.go.
when tests are done there is info "STEP: Destroying namespace "e2e-tests-horizontal-pod-autoscaling-xxxxx" for this suite"

File: test/e2e/autoscaling_utils.go (hunk 2 above)
Comment (socaa) [inline test/e2e/autoscaling_utils.go]: yes. sorry

File: test/e2e/autoscaling_utils.go (hunk 3 above)
Comment (socaa) [inline test/e2e/autoscaling_utils.go]: Deletion of experimental objects when Namespace is removed. #13553

Comment (mwielgus) [inline test/e2e/autoscaling_utils.go]: Ok, so if the namespace is deleted why to bother about RC and service?

Comment (piosz) [inline test/e2e/autoscaling_utils.go]: The service may consume resource so you may want to stop the consumption explicitly.

**Comment by piosz** [comment]
@socaa could you please rebase this PR?

**Comment by socaa** [comment]
@piosz done

**Comment by piosz** [comment]
LGTM

**Comment by piosz** [comment]
@k8s-bot test this please
